File: chroma-league/src/main/java/com/bonepl/chromaleague/tasks/FetchGameStatsTask.java
package com.bonepl.chromaleague.tasks;

import com.bonepl.chromaleague.rest.http.LeagueHttpClients;
import com.bonepl.chromaleague.rest.http.handlers.GameStatsResponseHandler;
import com.bonepl.chromaleague.state.RunningState;

import java.util.logging.Logger;

public class FetchGameStatsTask implements Runnable {
    public static final String URL = "https://127.0.0.1:2999/liveclientdata/gamestats";
    public static final String DEFAULT_MAP_TERRAIN = "Default";
    private static final Logger LOGGER = Logger.getLogger(FetchGameStatsTask.class.getName());
    private final GameStatsResponseHandler gameStatsResponseHandler = new GameStatsResponseHandler();

    @Override
    public void run() {
        LeagueHttpClients.getNonBlockingResponse(URL, gameStatsResponseHandler)
                .ifPresent(gameStats -> RunningState.getGameState().setGameStats(gameStats));
    }
}

File: src/tasks/FetchGameStatsTask.ts
import {getNonBlockingResponse} from '../rest/http/LeagueHttpClients.js';
import type {GameStats} from '../rest/gamestats/GameStats.js';
import {RunningState} from '../state/RunningState.js';

const URL = 'https://127.0.0.1:2999/liveclientdata/gamestats';

export async function fetchGameStatsTask(): Promise<void> {
    const gameStats = await getNonBlockingResponse<GameStats>(URL);
    if (gameStats) {
        RunningState.getGameState().gameStats = gameStats;
    }
}
